app/models/node.rb

```diff
@@ -0,0 +1,8 @@
+class Node < ApplicationRecord
+	belongs_to :parent, class_name: 'Node', required: false
+	has_many :children, :class_name => 'Node', :foreign_key => 'parent_id'
+
+	def self.root
+		self.find_by(parent_id: nil)
```

Comment:
As the method is defined in the scope of the class, it will execute in that scope. You don't need to write self.find_by.